### PR TITLE
submit chall01 solution

### PR DESCRIPTION
**Pull request by tdawson-42**
First I check the amount of arguments is correct and that the input is a string that is at least 1 character long. Then I create a dictionary with matching letter : morse code pairs. Case doesn't matter with morse code so we make the string all lowercase. I then check that the string contains only characters that are keys in the dictionary (a-z and space).  We can then iterate through the string one character at a time and print the morse code equivalent for each character.

### CODE REVIEW

File: chall01/tdawson.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/python3
+import sys
+
+if len(sys.argv) != 2 or len(sys.argv[1]) == 0:
```
Comment (jgengo) [inline chall01/tdawson.py:4]:
```python
len(sys.argv[1]) == 0
```
can be written
```python
or not sys.argv[1]
```

File: chall01/tdawson.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    sys.exit(f'usage: {sys.argv[0]} <a-zA-Z string>')
```
Comment (jgengo) [inline chall01/tdawson.py:5]: really good! really!

File: chall01/tdawson.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+morse_alphabet = {'a': '.-', 'b': '-...', 'c': '-.-.', 'd': '-..', 'e': '.',
```
Comment (jgengo) [inline chall01/tdawson.py:7]: once again really good!

File: chall01/tdawson.py [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+                  'f': '..-.', 'g': '--.', 'h': '....', 'i': '..', 'j': '.---',
+                  'k': '-.-', 'l': '.-..', 'm': '--', 'n': '-.', 'o': '---', 'p': '.--.',
+                  'q': '--.-', 'r': '.-.', 's': '...', 't': '-', 'u': '..-', 'v': '...-',
+                  'w': '.--', 'x': '-..-', 'y': '-.--', 'z': '--..', ' ': ' '}
+
+string = sys.argv[1].lower()
+
+if not all(c in morse_alphabet for c in string):
+    sys.exit(f'usage: {sys.argv[0]} <a-zA-Z string>')
```
Comment (jgengo) [inline chall01/tdawson.py:16]: good!

File: chall01/tdawson.py [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+for c in string:
+    print(morse_alphabet[c], end='')
```
Comment (jgengo) [inline chall01/tdawson.py:19]: I would suggest avoiding printing char by char but better to append into a string and then print the string at the end.

Comment (tdawson-42) [reply chall01/tdawson.py:19]: Hi, thank you for your feedback, it is very much appreciated! 👍 

I agree with your suggestion here, I learned about the `String.join()` method so I would now do it this way:
```Python
print(''.join(morse_alphabet[c] for c in string))
```

**Pull request merged**